File: QadeerApp.Web/Modules/Cv/EmployeeCv/EmployeeCvColumns.cs
using Serenity.ComponentModel;

namespace QadeerApp.Cv.Columns;

[ColumnsScript("Cv.EmployeeCv")]
[BasedOnRow(typeof(EmployeeCvRow), CheckNames = true)]
public class EmployeeCvColumns
{
    [EditLink, Width(200)]
    public string UserDisplayName { get; set; }

    [Width(140)]
    public string EmployeeNumber { get; set; }

    [Width(140)]
    public string Mobile { get; set; }

    [Width(160)]
    public EnglishLevel EnglishLevel { get; set; }

    [Width(120)]
    public string ExtensionNumber { get; set; }

    [Width(120)]
    public string OfficeNumber { get; set; }

    [Width(120)]
    public string BuildingNumber { get; set; }

    [DisplayName("Last Updated"), Width(150)]
    public DateTime CvUpdatedOn { get; set; }
}

File: QadeerApp.Web/Modules/ServerTypes/Cv/EmployeeCvColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { EmployeeCvRow } from "./EmployeeCvRow";
import { EnglishLevel } from "./EnglishLevel";

export interface EmployeeCvColumns {
    UserDisplayName: Column<EmployeeCvRow>;
    EmployeeNumber: Column<EmployeeCvRow>;
    Mobile: Column<EmployeeCvRow>;
    EnglishLevel: Column<EmployeeCvRow>;
    ExtensionNumber: Column<EmployeeCvRow>;
    OfficeNumber: Column<EmployeeCvRow>;
    BuildingNumber: Column<EmployeeCvRow>;
    CvUpdatedOn: Column<EmployeeCvRow>;
}

export class EmployeeCvColumns extends ColumnsBase<EmployeeCvRow> {
    static readonly columnsKey = 'Cv.EmployeeCv';
    static readonly Fields = fieldsProxy<EmployeeCvColumns>();
}

[EnglishLevel]; // referenced types